components/ContractViewer/DeploymentInfo.ts

```diff
@@ -149,13 +150,35 @@ export class DeploymentInfo {
     }
   }
 
-  async compile(outputs: string[], version?: string) {
+  async compile(
+    outputs: string[],
+    settings?: SoliditySettings,
+    version?: string,
+  ) {
+    if (!settings) {
+      settings = {
+        outputSelection: {
+          '*': {
+            '': outputs,
+            '*': outputs,
+          },
+        },
+      }
+    }
+
+    settings.outputSelection['*'][''] = (
+      settings.outputSelection['*'][''] || []
+    ).concat(outputs)
+    settings.outputSelection['*']['*'] = (
+      settings.outputSelection['*']['*'] || []
+    ).concat(outputs)

```

Comment:
nit, but shouldn't this be part of a else statement ? 

In the case there are no settings, we will create a `outputSelection` with `output` and then concat it again. (this doesn't break anything, but made me have read the code twice to make sure I understood it) 

```suggestion
else {
    settings.outputSelection['*'][''] = (
      settings.outputSelection['*'][''] || []
    ).concat(outputs)
    settings.outputSelection['*']['*'] = (
      settings.outputSelection['*']['*'] || []
    ).concat(outputs)
}
```

Reply:
Oh my bad, small typo. I meant to initialize `outputSelection` to empty arrays instead of to `output`. In that case it doesn't need to be part of the `else` statement.